Tests/TestReferenceTypes/FixtureReferenceTypes.swift

```diff
@@ -0,0 +1,87 @@
+import MockoloFramework
+
+let referenceTypesWithWeakAnnotation = """
+/// \(String.mockAnnotation)(referenceType: listener = weak)
+protocol Foo {
+    var listener: AnyObject? { get }
+    func barFunc(val: [Int])
+    func bazFunc(arg: String, other: Float)
+}
+"""
+
+let referenceTypesWithWeakAnnotationMock = """
+class FooMock: Foo {
+    init() { }
+    init(listener: AnyObject? = nil) {
+        self.listener = listener
+    }
+
+
+    private(set) var listenerSetCallCount = 0
+    weak var listener: AnyObject? = nil { didSet { listenerSetCallCount += 1 } }
+
+    private(set) var barFuncCallCount = 0
+    var barFuncHandler: (([Int]) -> ())?
+    func barFunc(val: [Int])  {
+        barFuncCallCount += 1
+        if let barFuncHandler = barFuncHandler {
+            barFuncHandler(val)
+        }
+
+    }
+
+    private(set) var bazFuncCallCount = 0
+    var bazFuncHandler: ((String, Float) -> ())?
+    func bazFunc(arg: String, other: Float)  {
+        bazFuncCallCount += 1
+        if let bazFuncHandler = bazFuncHandler {
+            bazFuncHandler(arg, other)
+        }
+
+    }
+}
+
+"""
+
+let referenceTypesWithDynamicAnnotation = """
+/// \(String.mockAnnotation)(referenceType: listener = dynamic)
```

Comment:
Add a test case for a dynamic func also. Please verify the generated mocks are valid (try compiling them in source code).